ps-modules/MacUtil/MacUtil.psm1

```diff
@@ -0,0 +1,202 @@
+Import-Module "$PSScriptRoot/../../ps-modules/Util"
+
+function ConvertTo-UniversalBinaries {
+    param (
+        [string]$arm64Dir,
+        [string]$x64Dir,
+        [string]$universalDir
+    )
+
+    Write-Host "Creating universal dir and copying headers..."
+    if (Test-Path -Path $universalDir -PathType Container) {
+       Remove-Item -Path "$universalDir" -Recurse -Force | Out-Null
+    }
+    New-Item -Path "$universalDir" -ItemType Directory -Force | Out-Null
+    $arm64LibDir = Join-Path $arm64Dir "lib"
+    $x64LibDir = Join-Path $x64Dir "lib"
+    Copy-Item -Path "$x64Dir\include" -Destination "$universalDir\include" -Recurse | Out-Null # Assume arm64 and x86_64 are identical
+
+    Write-Host "Making install paths relative..."
+    ConvertTo-RelativeInstallPaths -directory $arm64LibDir -extension "a"
```

Comment:
Does this really belong here? Seems like a major operation that shouldn't be inside of "ConvertTo-UnviersalBinaries" and violating single responsibility.

Reply:
I kind of like having the "do the mac-specific stuff afterwards to make the artifacts ready to stage" in one function to call from `build-and-stage.ps1`.  What do you think about refactoring it to something like this to kind of break out the different steps a bit better?

- Create-FinalMacArtifacts ($srcX64Dir, $srcArm64Dir, $destUniversalDir)
  - ...copy headers...
  - Update-LibsToRelativePaths ($srcX64Dir, $srcArm64Dir)
  - Create-UniversalBinaries ($srcX64Dir, $srcArm64Dir, $destUniversalDir)

Reply:
What do you think about this? 528293e32babb6e85d111b3bcc40d7f1b7a60b38